GrandCentralBoard/Core/Extensions/UIView.swift

```diff
@@ -29,22 +29,65 @@ public extension UIView {
     }
 }
 
+/// () -> Void cannot be stored as associatedObject, we need a wrapper.
+private final class BlockWrapper {
+    let block: () -> Void
+    
+    init(block: () -> Void) {
+        self.block = block
+    }
+}
+
 public extension UIView {
+    private struct AssociatedKey {
+        static var onDidBecomeActiveKey = "onDidBecomeActiveKey"
+    }
+    
+    private var onApplicationDidBecomeActive: (() -> Void)? {
+        get {
+            let wrapper = objc_getAssociatedObject(self, &AssociatedKey.onDidBecomeActiveKey) as? BlockWrapper
```

Comment:
You don't have to use Associated Objects to achieve this result.

> In the right hands, the functions of <objc/runtime.h> have the potential to add powerful new behavior to an application or framework, in ways that would otherwise not be possible. In the wrong hands, it drains the proverbial sanity meter of the code, and everything it may interact with (with terrifying side-effects).

>Therefore, it is with great trepidation that we consider this Faustian bargain, and look at one of the subjects most-often requested by NSHipster readers: associated objects.

>http://nshipster.com/associated-objects/